ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -52,62 +74,116 @@ static void propagate_layout(const std::shared_ptr<Node>& src, const std::shared
+                                "RGB/BGR color format using 'convert_color'");
+                if (nodes[0]->get_element_type().is_dynamic()) {
                     throw ngraph::ngraph_error("Can't insert 'convert_element_type' for dynamic source tensor type.");
                 }
-                auto convert = std::make_shared<op::v0::Convert>(node, type);
-                convert->set_friendly_name(node->get_friendly_name() + "/convert_element_type");
-                propagate_layout(node, convert);
+                auto convert = std::make_shared<op::v0::Convert>(nodes[0], type);
+                convert->set_friendly_name(nodes[0]->get_friendly_name() + "/convert_element_type");
+                propagate_layout(nodes[0], convert);
                 return convert;
             },
             true));
     }
-    std::list<std::tuple<PreProcessSteps::CustomPreprocessOp, bool>> m_actions;
+
+    void add_convert_color_impl(const ColorFormat& dst_format) {
+        m_actions.emplace_back(std::make_tuple(
+            [&, dst_format](const std::vector<std::shared_ptr<Node>>& nodes, PreprocessingContext& context) {
+                auto get_nv12_conv_format = [](const ColorFormat& dst_format) -> op::v8::ConvertColorNV12::ColorFormat {
+                    switch (dst_format) {
+                    case ColorFormat::RGB:
+                        return op::v8::ConvertColorNV12::ColorFormat::NV12_TO_RGB;
+                    case ColorFormat::BGR:
+                        return op::v8::ConvertColorNV12::ColorFormat::NV12_TO_BGR;
+                    default:
+                        OPENVINO_ASSERT(false, "Unsupported NV12 conversion format");
+                    }
+                };
+                if (context.m_color_format == ColorFormat::NV12_SINGLE_PLANE) {
+                    OPENVINO_ASSERT(nodes.size() == 1,
+                                    "Internal error: single plane NV12 image can't have multiple inputs");
+                    auto conversion_format = get_nv12_conv_format(dst_format);
+                    auto convert = std::make_shared<op::v8::ConvertColorNV12>(nodes[0], conversion_format);
+                    convert->set_friendly_name(nodes[0]->get_friendly_name() + "/convert_color_nv12_single");
+                    context.m_color_format = dst_format;
+                    return convert;
+                } else if (context.m_color_format == ColorFormat::NV12_TWO_PLANES) {
+                    OPENVINO_ASSERT(nodes.size() == 2,
+                                    "Internal error: two-plane NV12 image must have exactly two inputs");
+                    auto conversion_format = get_nv12_conv_format(dst_format);
+                    auto convert = std::make_shared<op::v8::ConvertColorNV12>(nodes[0], nodes[1], conversion_format);
+                    convert->set_friendly_name(nodes[0]->get_friendly_name() + "/convert_color_nv12_two_planes");
```

Comment:
looks like we need to add tensor names here like `orig_name/y` and `orig_name/uv`?

Reply:
Added tests for this